### ISSUE TITLE
Can flux.jl handle networks with more than one input?

### ISSUE THREAD

**Issue by thvhauwe** [FluxML/Flux.jl#452]
For example a network with two distinct inputs in Keras: https://keras.io/getting-started/functional-api-guide/#multi-input-and-multi-output-models

I'm getting errors when running the following basic example code trying to train a network with two inputs and 1 output. The network is made up out of a left branch and a right branch

    using Flux

    nINPUTS=1024
    nHIDDEN=353
    nOUTPUTS=54

    W1 = param(rand(nHIDDEN, nINPUTS))
    b1 = param(rand(nHIDDEN))
    layer1(x) = W1 * x .+ b1

    W2 = param(rand(nOUTPUTS, nHIDDEN))
    b2 = param(rand(nOUTPUTS))
    layer2(x) = W2 * x .+ b2

    left_branch(x) = layer2(relu.(layer1(x)))

    W3 = param(ones(nOUTPUTS,nOUTPUTS))
    b3 = param(zeros(nOUTPUTS))
    layer3(y) = W3 * y .+ b3

    right_branch(y)=layer3(y)

    model(A,B)=left_branch(A) .+ right_branch(B)

    da,db,dc = rand(nINPUTS),ones(nOUTPUTS),rand(nOUTPUTS)

    Zipped_Data = zip(da,db,dc)

    loss(x,y,z) = Flux.mse(model(x,y),z)

    optSGD=Flux.Optimise.SGD([W1,W2,W3,b1,b2,b3], η = 0.01)

    Flux.train!(loss,Zipped_Data,optSGD)

**Comment by PastelBelem8** [comment]
> I'm getting errors when running the following basic example code trying to train a network with two inputs and 1 output. The network is made up out of a left branch and a right branch

To train a network with 2 input values and 1 output value, you have to reflect that in the number of inputs `nINPUTS` (which is now saying that your network is expecting 1024 inputs) and in the `nOUTPUTS` (which is now saying that your network will output a vector with 54 values). 

I suggest changing the values to 

```
nINPUTS=2
nHIDDEN=353
nOUTPUTS=1
```
After that, since SGD is exported by the Flux's submodule Optimise you can directly call
`optSGD=Flux.SGD([W1,W2,W3,b1,b2,b3], 0.01)`

I hope that helps, let me know if I can help in anything else (:

**Comment by PastelBelem8** [comment]
Also note that you can significantly simplify the network that you are trying to achieve. I believe the last layer it is not required, either way, you could optionally do something like the one below:
```julia
using Flux
nINPUTS=2
nHIDDEN=353
nOUTPUTS=1
# Create one input vector of size 2 and the corresponding target 
da, db = rand(nINPUTS), rand(nOUTPUTS)

# Create the data batch to be processed by the network
# The expected format for multiple samples is:
# [([x11, x12], y1), ([x21, x22], y2), ..., ([xn1, xn2], yn])
data = [(da, db[1])]

# Create the model 
model = Chain( Dense(nINPUTS, nHIDDEN, identity),
               Dense(nHIDDEN, nOUTPUTS, identity),
               Dense(nOUTPUTS, nOUTPUTS))

# Define the loss function to be used in backprop 
loss(x, y) = Flux.mse(model(x), y)

# Define the optimiser
prms = params(model) # retrieve the network params of the previously defined model
optSGD = Flux.SGD(prms, 0.01)

# Train it!
Flux.train!(loss, data, optSGD)

```

**Comment by MikeInnes** [comment]
Assuming you got this figure out.
